packages/lib/server/getLuckyUser.ts

```diff
@@ -302,7 +366,284 @@ export async function getLuckyUser<
+    const chosenRoute = routes?.find((route) => route.id === routingFormResponse.chosenRouteId);
+
+    if (chosenRoute && "attributesQueryValue" in chosenRoute) {
+      const parsedAttributesQueryValue = children1Schema.parse(chosenRoute.attributesQueryValue);
+
+      const attributesQueryValueWithLabel = (await getAttributesQueryValue())({
+        attributesQueryValue: chosenRoute.attributesQueryValue,
+        attributes: [attributeWithWeights],
+        response,
+        fields: routingFormResponse.form.fields as Fields,
+        getFieldResponse: getFieldResponse,
+      });
+
+      const parsedAttributesQueryValueWithLabel = children1Schema.parse(attributesQueryValueWithLabel);
+
+      if (parsedAttributesQueryValueWithLabel && parsedAttributesQueryValueWithLabel.children1) {
+        averageWeightsHosts = getAverageAttributeWeights(
+          allRRHosts,
+          parsedAttributesQueryValueWithLabel.children1,
+          attributeWithWeights
+        );
+      }
+
+      if (parsedAttributesQueryValue && parsedAttributesQueryValue.children1) {
+        fieldOptionData = getAttributesForVirtualQueues(
+          response,
+          parsedAttributesQueryValue.children1,
+          attributeWithWeights
+        );
+      }
+    }
+  }
+
+  if (fieldOptionData) {
+    return { averageWeightsHosts, virtualQueuesData: { chosenRouteId, fieldOptionData } };
+  }
+
+  return;
+}
+
+function getAverageAttributeWeights<
+  T extends PartialUser & {
+    priority?: number | null;
+    weight?: number | null;
+  }
+>(
+  allRRHosts: GetLuckyUserParams<T>["allRRHosts"],
+  attributesQueryValueChild: Record<
+    string,
+    {
+      type?: string | undefined;
+      properties?:
+        | {
+            field?: any;
+            operator?: any;
+            value?: any;
+            valueSrc?: any;
+          }
+        | undefined;
+    }
+  >,
+  attributeWithWeights: AttributeWithWeights
+) {
+  let averageWeightsHosts: { userId: number; weight: number }[] = [];
+
+  const fieldValueArray = Object.values(attributesQueryValueChild).map((child) => ({
+    field: child.properties?.field,
+    value: child.properties?.value,
+  }));
+
+  fieldValueArray.map((obj) => {
+    const attributeId = obj.field;
+    const allRRHostsWeights = new Map<number, number[]>();
+
+    if (attributeId === attributeWithWeights.id) {
+      obj.value.forEach((arrayobj: string[]) => {
+        arrayobj.forEach((attributeOption: string) => {
+          // attributeOption is either optionId or label, we only care about labels here
+          const attributeOptionWithUsers = attributeWithWeights.options.find(
+            (option) => option.value.toLowerCase() === attributeOption.toLowerCase()
+          );
+
+          allRRHosts.forEach((rrHost) => {
+            const weight = attributeOptionWithUsers?.assignedUsers.find(
+              (assignedUser) => rrHost.user.id === assignedUser.member.userId
+            )?.weight;
```

Comment:
I think we should have a default here. We won't be assigning weights to all members in the queue I believe.

Without adding a default getLuckyUser was crashing with the following error:

` Weight filter should never return length=0`